### PR TITLE
Feature/ Warn user when using an outdated version of the inference server

### PR DESCRIPTION
**Pull request by SolomonLake**
# Description

Adds a check for latest version in version.py, which prints a recurring warning if the current version doesn't match.

### CODE REVIEW

File: inference/core/version.py [hunk 1]
```diff
@@ -1,4 +1,43 @@
+import requests
+import time
+import threading
+from inference.core.logger import logger
+
 __version__ = "0.8.5"
 
+latest_release = None
```
Comment (SolomonLake) [inline inference/core/version.py:8]: All this code should probably live in a separate file, `check_version.py`, maybe in a utils folder? Then we can choose more easily when to print the warning. How often do you think we should print the message, every 5 minutes, on every infer?

What do you think @paulguerrie ?

Comment (paulguerrie) [reply inference/core/version.py:10]: I could be convinced either way. I think this is a fine spot for it.

**Review by paulguerrie — APPROVED** [review]
Looks great!

File: inference/core/version.py (hunk 1 above)
Comment (SolomonLake) [reply inference/core/version.py:10]: Ok, lets 🚢 it. We can update later as needed.

**Pull request merged**